src/lsm/k_way_merge.zig

```diff
@@ -63,7 +63,7 @@ pub fn KWayMergeIterator(
             // TODO Do we have test coverage for this edge case?
             var stream_index: u32 = 0;
             while (stream_index < stream_count_max) : (stream_index += 1) {
-                it.keys[it.k] = stream_peek(context, stream_index) orelse continue;
+                it.keys[it.k] = stream_peek(context, stream_index) catch continue;
```

Comment:
Couldn't it be `Pending` since init() is called after the iterators are tick()'d the first time in Compaction (so read values but may not be fully buffered in memory) ?

Reply:
Ah, i realized it's not peeking the same stream twice. And yes, we should have values buffered since its after the first tick()s.